source/js/main.js

```diff
@@ -146,6 +146,64 @@ let main = {
       MozfestCarousels.init();
       MozfestHeroCarousels.init();
     }
+
+    const today = new Date();
+
+    // skip the banner if it got dismissed by the user today already
+    if (parseInt(localStorage.getItem("DISMISS_KEY")) === today.getDay()) {
+      document.querySelector(`.donate-banner`).style.display = "none";
+    }
+
+    document
```

Comment:
there is no need to add this event listener if there is a dismissal key and the day hasn't rolled over yet, so let's add a short circuit here. E.g. 
```
const banner = document.querySelector(`.donate-banner`);
const hideBanner = parseInt(localStorage.getItem("DISMISS_KEY")) === today.getDay();
const closeButton = hideBanner? undefined : banner.querySelector(`a.banner-close`);

if(hideBanner) { 
  banner.classList.add(`tw-hidden`);
}

closeButton?.addEventListener(......
```